iroha-lib/model/generators/CommandGenerator.cpp

```diff
@@ -0,0 +1,258 @@
+#include "CommandGenerator.hpp"
+#include "model/converters/json_transaction_factory.hpp"
+
+
+namespace iroha_lib
+{
+
+std::shared_ptr<iroha::protocol::Command> CommandGenerator::generateAddAssetQuantity(const std::string& asset_id,
```

Comment:
Keep in mind that this returns a `shared_ptr` which usually means that there's an atomic reference count being incremented.